ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -424,10 +430,15 @@ func (r *RayServiceReconciler) reconcileRayCluster(ctx context.Context, rayServi
 	if clusterAction == RolloutNew {
 		// For LLM serving, some users might not have sufficient GPU resources to run two RayClusters simultaneously.
 		// Therefore, KubeRay offers ENABLE_ZERO_DOWNTIME as a feature flag for zero-downtime upgrades.
+		zeroDowntimeEnvVar := os.Getenv(ENABLE_ZERO_DOWNTIME)
+		rayServiceSpecUpgradeStrategy := rayServiceInstance.Spec.UpgradeStrategy
 		enableZeroDowntime := true
-		if s := os.Getenv(ENABLE_ZERO_DOWNTIME); strings.ToLower(s) == "false" {
-			enableZeroDowntime = false
+		if zeroDowntimeEnvVar != "" {
```

Comment:
What will happen if we set `ENABLE_ZERO_DOWNTIME: true` and `Spec.UpgradeStrategy: None`?

As discussed, `ENABLE_ZERO_DOWNTIME` is an operator-level configuration, while `Spec.UpgradeStrategy` is a CR-level configuration. The CR-level configuration should take priority over `ENABLE_ZERO_DOWNTIME`.

Reply:
Can you also write down some comments to summarize the mechanism to control zero-downtime upgrade as I mentioned https://github.com/ray-project/kuberay/pull/2468/files#r1823903220?

Reply:
Made it so that CR will take precedence if it isn't empty.

I added some comments right above `enableZeroDowntime` variable.

Reply:
thanks!